pkg/console/operator/sync_v400.go

```diff
@@ -125,21 +118,19 @@ func (co *consoleOperator) sync_v400(updatedOperatorConfig *operatorv1.Console,
 	klog.V(4).Infof("sync loop 4.0.0 resources updated: %v", toUpdate)
 	klog.V(4).Infoln("-----------------------")
 
-	// the operand is in a transitional state if any of the above resources changed
-	// or if we have not settled on the desired number of replicas or deployment is not up to date.
-	if toUpdate {
-		co.ConditionResourceSyncProgressing(updatedOperatorConfig, "Changes made during sync updates, additional sync expected.")
-	} else {
+	co.HandleProgressing(updatedOperatorConfig, "ResourceSyncUpdatesInProgress", func() error {
+		if toUpdate {
+			return errors.New("Changes made during sync updates, additional sync expected.")
+		}
 		version := os.Getenv("RELEASE_VERSION")
 		if !deploymentsub.IsAvailableAndUpdated(actualDeployment) {
-			co.ConditionResourceSyncProgressing(updatedOperatorConfig, fmt.Sprintf("Working toward version %s", version))
-		} else {
-			if co.versionGetter.GetVersions()["operator"] != version {
-				co.versionGetter.SetVersion("operator", version)
-			}
-			co.ConditionResourceSyncNotProgressing(updatedOperatorConfig)
+			return errors.New(fmt.Sprintf("Working toward version %s", version))
 		}
-	}
+		if co.versionGetter.GetVersions()["operator"] != version {
```

Comment:
I'm not 100% on this being hidden within the `func() {}()` of `HandleProgressing()`, but it works & is consistent with the previous code.  Its just a tad "dirty" as the callback func shouldn't really have side effects, it should simply return an error, or not.